src/code_80005FD0.c

```diff
@@ -4415,34 +4414,34 @@ void init_course_vehicles(void) {
     case COURSE_TOADS_TURNPIKE:
         for(loopIndex = 0; loopIndex < NUM_RACE_BOX_TRUCKS; loopIndex++) {
             tempBoxTruck = &gBoxTruckList[loopIndex];
-            func_80012220(tempBoxTruck);
+            init_vehicle_on_road(tempBoxTruck);
             tempBoxTruck->actorIndex = add_actor_to_empty_slot(tempBoxTruck->position, tempBoxTruck->rotation, tempBoxTruck->velocity, ACTOR_BOX_TRUCK);
         }
         for(loopIndex = 0; loopIndex < NUM_RACE_SCHOOL_BUSES; loopIndex++) {
             tempSchoolBus = &gSchoolBusList[loopIndex];
-            func_80012220(tempSchoolBus);
+            init_vehicle_on_road(tempSchoolBus);
             tempSchoolBus->actorIndex = add_actor_to_empty_slot(tempSchoolBus->position, tempSchoolBus->rotation, tempSchoolBus->velocity, ACTOR_SCHOOL_BUS);
         }
         for(loopIndex = 0; loopIndex < NUM_RACE_TANKER_TRUCKS; loopIndex++) {
             tempTankerTruck = &gTankerTruckList[loopIndex];
-            func_80012220(tempTankerTruck);
+            init_vehicle_on_road(tempTankerTruck);
             tempTankerTruck->actorIndex = add_actor_to_empty_slot(tempTankerTruck->position, tempTankerTruck->rotation, tempTankerTruck->velocity, ACTOR_TANKER_TRUCK);
         }
         for(loopIndex = 0; loopIndex < NUM_RACE_CARS; loopIndex++) {
             tempCar = &gCarList[loopIndex];
-            func_80012220(tempCar);
+            init_vehicle_on_road(tempCar);
             tempCar->actorIndex = add_actor_to_empty_slot(tempCar->position, tempCar->rotation, tempCar->velocity, ACTOR_CAR);
         }
         break;
     }
 }
 
-void func_80012780(TrainCarStuff *trainCar, PathNoY *arg1, u16 arg2) {
-    trainCar->position[0] = (f32) arg1->x;
+void set_vehicle_pos_waypoint(TrainCarStuff *trainCar, s16 *posXZ, u16 waypoint) {
```

Comment:
s16 *posXZ

Shouldn't this be your new Path2D struct? and just pos instead of posXZ?